common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerRequest.java

```diff
@@ -292,4 +298,19 @@ public void setSdkType(SdkType sdkType) {
         this.mSdkType = sdkType;
     }
 
+
+    public static String getJsonStringForExtraQueryParams(List<Pair<String, String>> mExtraQueryStringParameters) {
```

Comment:
I might prefer to see this factored slightly differently:
- These two `static` methods can move to `QueryParamsAdapter` (`getJsonStringForExtraQueryParams`, `getExtraQueryParamsFromJsonString`)
- You can initialize the `gson` instance once [eagerly] on the object
- Consider a name like `QueryParamsAdapter._toJson()` and `QueryParamsAdapter._fromJson()` (The `_` prefix is to avoid instance method name collisions)

Roughly

```java
public class QueryParamsAdapter extends TypeAdapter<List<Pair<String, String>>> {

    private static final Gson mGson;

    static {
        final GsonBuilder gsonBuilder = new GsonBuilder();
        gsonBuilder.registerTypeAdapter(
                QueryParamsAdapter.class,
                new QueryParamsAdapter()
        );
        mGson = gsonBuilder.create();
    }

    // ... Other methods

    public static String _toJson(List<Pair<String, String>> extraQueryStringParameters) {
        return mGson.toJson(mExtraQueryStringParameters, QueryParamsAdapter.class);
    }

    public static List _fromJson(String jsonString) {
        return (List<Pair<String, String>>) mGson.fromJson(jsonString, QueryParamsAdapter.class);
    }
}
```